Tools/autotest/arduplane.py

```diff
@@ -1327,6 +1327,46 @@ def airspeed_autocal(self):
         self.wait_waypoint(7, num_wp-1, timeout=500)
         self.wait_disarmed(timeout=120)
 
+    def deadreckoning_main(self, disable_airspeed_sensor=False):
+        # wind is from the West:
+        self.set_parameter("SIM_WIND_DIR", 270)
+        # light winds:
+        self.set_parameter("SIM_WIND_SPD", 10)
+        if disable_airspeed_sensor:
+            self.set_parameter("ARSPD_USE", 0)
+
+        self.takeoff(50)
+        loc = self.mav.location()
+        loc.lat = -35.35690712
+        loc.lng = 149.17083386
+        self.run_cmd_int(
+            mavutil.mavlink.MAV_CMD_DO_REPOSITION,
+            0,
+            mavutil.mavlink.MAV_DO_REPOSITION_FLAGS_CHANGE_MODE,
+            0,
+            0,
+            int(loc.lat*1e7),
+            int(loc.lng*1e7),
+            100,    # alt
+            frame=mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT_INT,
+        )
+        self.wait_location(loc, accuracy=100)
+        self.progress("Stewing")
+        self.delay_sim_time(20)
+        self.set_parameter("SIM_GPS_DISABLE", 1)
+        self.progress("Roasting")
+        self.delay_sim_time(20)
+        self.change_mode("RTL")
+        self.wait_distance_to_home(100, 200, timeout=200)
```

Comment:
how does this test dead-reckoning? The state estimation could be completely off and it would pass. We'd have to check the SIMSTATE

Reply:
It does test the most important thing - that the plane flies home again.

You're right in that I should tighten the check to make sure that's not a fluke.